s3parq/fetch_parq.py

```diff
@@ -329,21 +332,26 @@ def _get_filtered_key_list(typed_parts: dict, filters, key) -> List[str]:
     are set ie all non-matching partitions are excluded.
     '''
     filter_keys = []
+    matched_parts = OrderedDict()
+    matched_parts.keys = typed_parts.keys()
+    matched_part_vals = set()
 
     for part, part_values in typed_parts.items():
-        for f in filters:
-            if (f['partition'] == part):
-                comparison = OPS[f['comparison']]
-                for v in f['values']:
-                    typed_parts[part] = set(filter(
-                        lambda x: comparison(x, v),
-                        typed_parts[part]
-                    )
-                    )
-
-    def construct_paths(typed_parts, previous_fil_keys: List[str]) -> None:
-        if len(typed_parts) > 0:
-            part = typed_parts.popitem(last=False)
+        matched_part_vals.clear()
+        fil = next((f for f in filters if f['partition'] == part), False)
+        if fil:
+            comparison = OPS[fil['comparison']]
+            for v in fil['values']:
```

Comment:
is this always going to have a `values`?  if we can't promise that and we don't want it to fail if it's missing values, we can use `for v in fil.get('values', [])`.  the `get` will default to an empty array if `values` is missing.

if when `values` is missing something is clearly not right and we'd prefer for it to fail with a `KeyError`, i think this is ok

Reply:
Each filter requires a partition, a series of values, and a comparison operator. This is the defined schema for filters called through s3parq. If a partition is found for a particular filter, I think it is a fair assumption for the filter to contain a list of values and an operator. The original code also reflected this assumption.